File: CivitasJava/civitas/CasillaJuez.java
package civitas;

import java.util.ArrayList;

public class CasillaJuez extends Casilla {
    static private int carcel;
    
    CasillaJuez(int numCasillaCarcel, String s){
        super(s);
        carcel = numCasillaCarcel;
    }
    
    @Override
    void recibeJugador (int iactual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(iactual,todos)){
            informe(iactual,todos);
            todos.get(iactual).encarcelar(carcel);
        }
    }
}

File: CivitasRuby/casilla_juez.rb
require_relative 'casilla'

module Civitas
class CasillaJuez < Casilla
  def initialize(carcel, nombre)
    super(nombre)
    @@numCasillaCarcel = carcel 
  end
  
  def recibe_jugador(actual, todos)
    if jugador_correcto(actual, todos)
      informe(actual, todos)
      todos[actual].encarcelar(@@numCasillaCarcel)
    end
  end
  
end
end
